packages/database/data-migrations/20240611153820_pricing_v2/data-migration.ts

```diff
@@ -0,0 +1,172 @@
+import { Prisma, PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+type TSubscriptionStatusLegacy = "active" | "cancelled" | "inactive";
+
+interface TSubscriptionLegacy {
+  status: TSubscriptionStatusLegacy;
+  unlimited: boolean;
+}
+
+interface TFeatures {
+  inAppSurvey: TSubscriptionLegacy;
+  linkSurvey: TSubscriptionLegacy;
+  userTargeting: TSubscriptionLegacy;
+  multiLanguage: TSubscriptionLegacy;
+}
+
+interface TOrganizationBillingLegacy {
+  stripeCustomerId: string | null;
+  features: TFeatures;
+}
+
+async function main() {
+  await prisma.$transaction(
+    async (tx) => {
+      //   const startTime = Date.now();
+      console.log("Starting data migration for pricing v2...");
+
+      // Free tier
+      const orgsWithoutBilling = await tx.organization.findMany({
+        where: {
+          billing: {
+            path: ["stripeCustomerId"],
+            equals: Prisma.AnyNull,
+          },
+        },
+      });
+
+      console.log(
+        `Found ${orgsWithoutBilling.length} organizations without billing information. Moving them to free plan...`
+      );
+
+      for (const organization of orgsWithoutBilling) {
+        await tx.organization.update({
+          where: {
+            id: organization.id,
+          },
+          data: {
+            billing: {
+              stripeCustomerId: null,
+              plan: "free",
+              limits: {
+                monthly: {
+                  responses: 500,
+                  miu: 1000,
+                },
+              },
+            },
+          },
+        });
+      }
+
+      console.log("Moved all organizations without billing to free plan");
+
+      const orgsWithBilling = await tx.organization.findMany({
+        where: {
+          billing: {
+            path: ["stripeCustomerId"],
+            not: Prisma.AnyNull,
+          },
+        },
+      });
+
+      console.log(`Found ${orgsWithBilling.length} organizations with billing information`);
+
+      for (const org of orgsWithBilling) {
+        const billing = org.billing as TOrganizationBillingLegacy;
+
+        // @ts-expect-error
+        if (billing.plan && typeof billing.plan === "string") {
```

Comment:
This check can also make sense for the migration of users without billing information